### PR TITLE
ENT-2858 | Created an endpoint to link a user to an enterprise customer.

### PR DESCRIPTION
**Pull request by hasnain-naveed**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

File: enterprise/api/v1/views.py [hunk 1]
```diff
@@ -619,3 +620,57 @@ def post(self, request):
...
+
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+
+        if enterprise_customer is None:
+            raise LinkUserRequestError(
+                "There is no enterprise customer with the following UUID [{enterprise_customer_uuid}]".format(
+                    enterprise_customer_uuid=enterprise_customer_uuid
+                )
+            )
+        if email:
```
Comment (irfanuddinahmad) [inline enterprise/api/v1/views.py:651]: Check for None and validate.

Comment (hasnain-naveed) [reply enterprise/api/v1/views.py]: Already added. TYPO

File: enterprise/api/v1/views.py [hunk 2]
```diff
@@ -619,3 +621,57 @@ def post(self, request):
                 {'error': str('Request codes email could not be sent')},
                 status=HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+
+class LinkUserView(APIView):
+    """
+    API to link the user to an enterprise customer.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
```
Comment (muhammad-ammar) [inline enterprise/api/v1/views.py:630]: @hasnain-naveed Can we make this endpoint only available to service users?

Comment (muhammad-ammar) [reply enterprise/api/v1/views.py:630]: would be great if we can add a test too.

**Comment by hasnain-naveed** [comment]
No need for this PR. We already have an endpoint for this use case.

**Pull request closed without merging**